### ISSUE TITLE
Keep alive pings don't seem to work when there are a lot of idle connections

### ISSUE THREAD

**Issue by cdegroot** [elixir-ecto/db_connection#202]
It smells like these pings are run round-robin, not in parallel and only once per `idle_interval`, If you have a short wait_timeout and a bunch of idle connections, this means that connections can get lost (we use TLS, which makes losing connections quite expensive). 

The expectation is that each connection keeps its own timer and keeps itself, as it were, independent of other connections.

**Comment by josevalim** [comment]
Yes, I can confirm that's how it works currently. I think we can improve this by making the current detection recursive. Instead of only pinging a single connection, we will ping all connections unused in the idle interval. Here is the code to be made recursive:

https://github.com/elixir-ecto/db_connection/blob/master/lib/db_connection/connection_pool.ex#L122-L131

Here is where you can add a new test:

https://github.com/elixir-ecto/db_connection/blob/d78589549ca45ef5962bcf8a48fa32510121464a/integration_test/cases/idle_test.exs

I know the tests are very cryptic (IMO, the hardest part of maintaining this lib), so if help is necessary to write the test, please let me know.

**Comment by josevalim** [comment]
Ping. Any news here?

**Comment by cdegroot** [comment]
From whom? Nothing from me in any case :)

**Comment by josevalim** [comment]
I may have gotten the wrong impression but given you mentioned you were looking for ways to help, I was under the expectation that you would send a pull request to address the current behaviour?

**Comment by cdegroot** [comment]
I’m looking to see how we can help out here, but so far no capacity. Still working on that :)

**Comment by josevalim** [comment]
Ping?

**Comment by cdegroot** [comment]
Sorry, same status still (also: summer. That time when people are away all
the time to lie on beaches and stuff ;)).

**Comment by josevalim** [comment]
Well, it was also summer when you joined the discussion. :) So if summer was not a blocker for "evaluating" the project back then, I was hoping it would not get in the way of supporting the project, especially when you recognized there are areas for contributions.

**Comment by josevalim** [comment]
Closing this in favor #206.
